### PR TITLE
port dom.js and lang.js from webcomponents.js

### PR DESCRIPTION
**Pull request by notwaldorf**
👀 @azakus @sorvell @kevinpschaaf

- I copied `lang.js` and `dom.js` from `webcomponents/webcomponentsjs` to here, and nuked out of date things (`Function.bind`, `window.performance.now`, `rAF`)
- I don't know how to generate the `.travis.yml` file, but ran the 4 tests locally! 

Let me know if there's enough words in the Readme, or we need more (or less). 💪

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+# webcomponents-platform
+A set of basic platform polyfills needed by webcomponents.js, such as a
+constructable `CustomEvent` and basic ES6 shims (`Object.assign` and `Array.from`)
+
+## Browser Support
+
+The polyfills are intended to work in the latest versions of evergreen browsers. See below
+for our complete browser support matrix:
+
+| Polyfill   | IE10 | IE11+ | Chrome* | Firefox* | Safari 7+* | Chrome Android* | Mobile Safari* |
```
Comment (dfreedm) [inline README.md]: Remove IE 10 column

Comment (dfreedm) [reply README.md]: Also, Safari 9+

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# License
+
+Everything in this repo is BSD style license unless otherwise specified.
+
+Copyright (c) 2015 The Polymer Authors. All rights reserved.
```
Comment (dfreedm) [inline LICENSE.md]: 2016

File: package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "license": "BSD-3-Clause",
+  "bugs": {
+    "url": "https://github.com/webcomponents/webcomponents-platform/issues"
+  },
+  "scripts": {
+    "test": "wct"
+  },
+  "homepage": "http://webcomponents.org",
+  "devDependencies": {
+    "web-component-tester": "^4.0.1"
```
Comment (dfreedm) [inline package.json]: WCT 5+ please

**Review by dfreedm — CHANGES_REQUESTED** [review]

**Comment by dfreedm** [comment]
Let's not bother making this PR pass tests yet.

**Comment by notwaldorf** [comment]
@azakus done did changes, tests are still 😭

**Comment by notwaldorf** [comment]
@azakus LOOK AT DEM GREEN TESTS

**Review by dfreedm — APPROVED** [review]

**Comment by dfreedm** [comment]
👍

**Pull request merged**